File: app/src/main/java/com/example/lightshop/SidebarAdapter.java
package com.example.lightshop;

import android.graphics.Color;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.example.lightshop.models.CategoryModel;
import com.google.android.material.card.MaterialCardView;
import java.util.List;

public class SidebarAdapter extends RecyclerView.Adapter<SidebarAdapter.ViewHolder> {

    private final List<CategoryModel> categories;
    private final OnItemClickListener listener;
    private int selectedPosition = 0;

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    public SidebarAdapter(List<CategoryModel> categories, OnItemClickListener listener) {
        this.categories = categories;
        this.listener = listener;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public final View container;
        public final View selectionIndicator;
        public final MaterialCardView iconContainer;
        public final ImageView icon;
        public final TextView name;

        public ViewHolder(View view) {
            super(view);
            container = view.findViewById(R.id.sidebar_container);
            selectionIndicator = view.findViewById(R.id.selection_indicator);
            iconContainer = view.findViewById(R.id.icon_container);
            icon = view.findViewById(R.id.sidebar_icon);
            name = view.findViewById(R.id.sidebar_name);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_sidebar, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        CategoryModel category = categories.get(position);
        holder.name.setText(category.getCategoryName());
        
        holder.icon.clearColorFilter();
        
        // Handle local resource images or URL
        String imgUrl = category.getCategoryImage();
        if (imgUrl != null && !imgUrl.isEmpty()) {
            if (imgUrl.startsWith("res:")) {
                try {
                    int resId = Integer.parseInt(imgUrl.substring(4));
                    holder.icon.setImageResource(resId);
                } catch (Exception e) {
                    holder.icon.setImageResource(R.drawable.ic_category);
                }
            } else {
                Glide.with(holder.itemView.getContext())
                        .load(imgUrl)
                        .placeholder(R.drawable.ic_category)
                        .error(R.drawable.ic_category)
                        .into(holder.icon);
            }
        } else {
            holder.icon.setImageResource(R.drawable.ic_category);
        }

        if (position == selectedPosition) {
            holder.container.setBackgroundColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.icon_blue_bg));
            holder.selectionIndicator.setVisibility(View.VISIBLE);
            holder.name.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.primary_blue));
            holder.name.setTypeface(null, Typeface.BOLD);
            holder.iconContainer.setCardBackgroundColor(android.content.res.ColorStateList.valueOf(Color.WHITE));
        } else {
            holder.container.setBackgroundColor(Color.TRANSPARENT);
            holder.selectionIndicator.setVisibility(View.GONE);
            holder.name.setTextColor(ContextCompat.getColor(holder.itemView.getContext(), R.color.secondary_gray));
            holder.name.setTypeface(null, Typeface.NORMAL);
            holder.iconContainer.setCardBackgroundColor(android.content.res.ColorStateList.valueOf(Color.WHITE));
        }

        holder.itemView.setOnClickListener(v -> {
            int oldPos = selectedPosition;
            selectedPosition = holder.getBindingAdapterPosition();
            if (oldPos != selectedPosition) {
                notifyItemChanged(oldPos);
                notifyItemChanged(selectedPosition);
                if (listener != null) {
                    listener.onItemClick(selectedPosition);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return categories != null ? categories.size() : 0;
    }
}

File: app/src/main/java/com/example/lightshop/SidebarAdapter.kt
package com.example.lightshop

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView

class SidebarAdapter(
    private val categories: List<Category>,
    private val onItemClick: (Int) -> Unit
) : RecyclerView.Adapter<SidebarAdapter.ViewHolder>() {

    private var selectedPosition = 0

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val container: View = view.findViewById(R.id.sidebar_container)
        val icon: ImageView = view.findViewById(R.id.sidebar_icon)
        val name: TextView = view.findViewById(R.id.sidebar_name)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_sidebar, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val category = categories[position]
        holder.name.text = category.name
        holder.icon.setImageResource(category.iconRes)

        if (position == selectedPosition) {
            holder.container.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, R.color.selected_bg))
            holder.name.setTextColor(ContextCompat.getColor(holder.itemView.context, R.color.primary))
            holder.icon.setColorFilter(ContextCompat.getColor(holder.itemView.context, R.color.primary))
        } else {
            holder.container.setBackgroundColor(Color.TRANSPARENT)
            holder.name.setTextColor(ContextCompat.getColor(holder.itemView.context, R.color.text_subtitle))
            holder.icon.setColorFilter(ContextCompat.getColor(holder.itemView.context, R.color.text_hint))
        }

        holder.itemView.setOnClickListener {
            val oldPos = selectedPosition
            selectedPosition = holder.adapterPosition
            notifyItemChanged(oldPos)
            notifyItemChanged(selectedPosition)
            onItemClick(selectedPosition)
        }
    }

    override fun getItemCount() = categories.size
}
